### PR TITLE
GUI - Automatically measuring TextBlock size

### PR DESCRIPTION
**Pull request by royibernthal**
Continuing this PR:
https://github.com/BabylonJS/Babylon.js/pull/2915

### CODE REVIEW

File: gui/src/controls/textBlock.ts [hunk 1]
```diff
@@ -10,6 +10,17 @@ module BABYLON.GUI {
...
         private _totalHeight: number;
+        private _resizeToFit: boolean = false;
+
+        get resizeToFit(): boolean {
+            return this._resizeToFit;
+        }
+
+        set resizeToFit(value: boolean) {
+            this._resizeToFit = value;
+            this._width.ignoreAdaptiveScaling = true;
```
Comment (deltakosh) [inline gui/src/controls/textBlock.ts]: So :) this should be turned on only if value === true

File: gui/src/controls/textBlock.ts [hunk 2]
```diff
@@ -165,10 +205,25 @@ module BABYLON.GUI {
...
             for (var line of this._lines) {
                 this._drawText(line.text, line.width, rootY, context);
                 rootY += this._fontOffset.height;
+
+                if (line.width > maxLineWidth) maxLineWidth = line.width;
+            }
+
+            if (this._resizeToFit) {
+                this.width = maxLineWidth + 'px';
+                this.height = this._fontOffset.height * this._lines.length + 'px';
```
Comment (deltakosh) [inline gui/src/controls/textBlock.ts]: this is where you can add this.paddingLeftInPixels and this.paddingRIghtInPixels

**Comment by royibernthal** [comment]
Here :)

What do you think about my last questions in the previous PR?

> It looks like width and height are still being adapted, even when I'm setting width for instance to a fixed height (to make sure the problem is not in the measuring). What am I missing now?
> 
> console.log('width', this.width); does return a fixed '100px' as expected when ignoring adaptive scaling, but visually it is still scaled for a reason I can't point out yet.
> 
> I guess what I'm seeing is just the canvas resizing and the text block being resized as well since it's part of it? **At any rate something is still not working there.**

**Comment by deltakosh** [comment]
The canvas will resize and thus the fontsize will resize accordingly:
https://www.babylonjs-playground.com/#6C5J7E

So in turn the update you did will update the width. 

Can you repro the issue you're seeing?

**Comment by royibernthal** [comment]
Try to run this PG along with this PR:
http://www.babylonjs-playground.com/#HFNBXS

The rectangle sets itself to the size of the text so we can see the borders easily, and as you can see something is not right here.

**Comment by deltakosh** [comment]
It works if you do not forget to set your rect as ignoreAdaptiveScaling as well:
https://www.babylonjs-playground.com/#HFNBXS#1

**Comment by royibernthal** [comment]
I had a feeling the problem will end up being with the rectangle =\

Thanks for guiding me through this, looks like it's ready for merge?

**Comment by deltakosh** [comment]
Yep!

**Comment by deltakosh** [comment]
Can you update the doc accordingly?

**Comment by royibernthal** [comment]
Sure, how and where? :)

**Comment by royibernthal** [comment]
Oh got it, on it

**Comment by deltakosh** [comment]
https://github.com/BabylonJS/Documentation/blob/master/content/overviews/Standard/Gui.md
:)

**Pull request merged**